File: add_ga.py
import os

ga_snippet = """
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-7GNHWDKNZW"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-7GNHWDKNZW');
</script>
"""

for filename in os.listdir('.'):
    if filename.endswith('.html') and not filename.startswith('temp'):
        with open(filename, 'r', encoding='utf-8') as f:
            content = f.read()
        
        if 'G-7GNHWDKNZW' not in content:
            # find </head> and insert before it
            if '</head>' in content:
                content = content.replace('</head>', ga_snippet + '</head>')
                with open(filename, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Adicionado GA em {filename}")

File: add_ga.js
const fs = require('fs');

const ga = `
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-7GNHWDKNZW"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-7GNHWDKNZW');
</script>
</head>`;

const files = fs.readdirSync('.').filter(f => f.endsWith('.html') && !f.startsWith('temp'));

files.forEach(f => {
  let content = fs.readFileSync(f, 'utf8');
  if(!content.includes('G-7GNHWDKNZW') && content.includes('</head>')) {
    content = content.replace('</head>', ga);
    fs.writeFileSync(f, content, 'utf8');
    console.log(`Added GA to ${f}`);
  }
});
